### DISCUSSION TITLE
Zustand docs and website team

### DISCUSSION THREAD

**Discussion by armaandh** [pmndrs/zustand#1033] [Ideas]
Taking a look at jotai's official docs website, the documentation is clear and very useful. The https://zustand.surge.sh/ website is pretty to look at...but does not offer anything useful or practical in terms of documentation (must go to github for documentation). I propose adopting jotai's wonderful documentation website for Zustand. In otherwords, making zustand's official website have docs available in it. What do you think?

**Comment by dai-shi** [comment]
anyone volunteer?

jotai website https://jotai.org is done by jotai maintainers,
and valtio website https://valtio.pmnd.rs is done by valtio website team (and I'm less involved.)
It would be nice to have a team like valtio to move forward.

btw, https://zustand.surge.sh/ is obsoleted link, the current url is https://zustand-demo.pmnd.rs/ .

https://docs.pmnd.rs/zustand/ is a place meant to have docs, but it's not maintained.
What I think is good is to organize `./docs` in this repo and configure https://docs.pmnd.rs/zustand/ to use it.
related issue: #1008

**Reply by matteofrana** [reply]
I think the first thing should be to organize the Docs content into Topics > Subtopics > Sections to put things in order and have good placeholders for the docs to write (or copy/paste). With a clear plan it would be much easier to collaborate on the docs.
IMHO it would be better to start from a new Docusaurus site, organize the content structure and then write, so that we have also the search capability.

What do you think about it?

**Reply by dai-shi** [reply]
We don't maintain two docs for repo and site. All docs stay in `./docs` (and `./readme.md`) in this repo.

We have two things to do:
1. Organize contents like you said, but in `./docs`.
2. Make a PR to use that contents in https://docs.pmnd.rs/zustand/.

I think you would be interested in 1. Unless anyone is interested in 2, I will be working on it.

**Reply by matteofrana** [reply]
Hi @dai-shi, yes, I am interested in organizing the content topics. 
I will look at how content can be structured using the current /docs folder.

**Reply by dai-shi** [reply]
> Make a PR to use that contents in https://docs.pmnd.rs/zustand/.

Done: https://github.com/pmndrs/website/pull/240

Now, https://docs.pmnd.rs/zustand is in sync with our `./docs`.

**Reply by dbritto-dev** [reply]
Hey folks, what can I do to help with the docs?

**Reply by dai-shi** [reply]
Hi @dbritto-dev ,
Can you take a look at the current `./docs` and ideate about how they should be organized?
I hope to hear from @matteofrana and discuss together, but ideas for discussion would be welcome anyway.

Also, #1008 is not yet completed, which you can work on if interested.

**Reply by dbritto-dev** [reply]
@dai-shi cool, I'll start working on #1008 for now

**Reply by chrisk-7777** [reply]
I'd love to contribute too. I can put some ideas around organization, but if there are any other focus points, do let me know

**Reply by dai-shi** [reply]
Thanks.

I guess what we would need to are:
1. ideate docs organization
2. fix some existing contents, and
3. add missing contents

and maybe also the project management of this documenting activity.

**Reply by chrisk-7777** [reply]
I've just gone through the above, the linked issues, and the docs page itself. 

I'm not sure how you would like to handle the project management side of this, perhaps one issue (tagged as "documentation") based on some of the key areas below?

If that works, I'm happy to add a few issues, label them, and add requirements. I'll self-assign as much as possible. 

Would that work?

# Key areas

---

## Issue 1 - Add top level structure
Currently "Introduction, Recipes, etc"
In my opinion, Jotai' top level structure is quite clear to a newcomer. The Zustand docs would probably be initially smaller as there are fewer integrations in Zustand (I think?)

Initially, I'm thinking something like the following. This is a first pass to get things approximately slotted into place. 

 - Introduction (or "Getting Started", similar to Vite ?)
 - Basics
   - Concepts: Some of the content from first section of readme.md
   - Comparison: Similar to Jotai, and some content from the Zustand introduction section
   - Examples: Similar to Jotai "Showcase". Personally I prefer "example", to me, "showcase" sounds like production sites using Zustand. Thoughts?
 - Guides
   - Updating nested state object values (from current Zustand docs)
   - Immutable state and merging (from current Zustand docs)
   - Recommended practices (from current Zustand docs - "Flux inspired practice")
   - Auto Generating Selectors (from current Zustand docs)
   - Practice with no store actions (from current Zustand docs)
   - TypeScript (from current Zustand docs)
   - Testing (from current Zustand docs)
 - Recipes
   - Split recipes into individual guides (so approx 10 pages here)
 - Integrations (I don't know if this is broad enough for the below, let me know)
   - 3rd party libraries (from current Zustand docs)
   - Persist middleware (from current Zustand docs)
   - Devtools (from current Zustand readme.md)
 - Previous versions
   - V3 reateContext from zustand/context (from current Zustand docs)
 - Migrations
   - V4 migrations (from current Zustand docs)

---


## Issue 2 - Missing images & dead links audit
There are missing images (like on the introduction page) and dead links (like on the v4 migration page). 
The person on this issue should (manually) scan through the docs pages looking for missing images or broken links. 
A single PR should be able to apply all the fixes

---

## Issue 3 - Grammar, spelling, and phrasing audit
[minor]
Similar to issue 2. Do a sweep of all pages for any glaring grammar or spelling mistakes. 
It's a minor thing, but it can be distracting in my opinion. 

---

## Issue 4 - Style fixes
[minor]
There are some minor style fixes that could be made. Examples:
 - The `<summary>` child content isn't indented / nested so is lost in the body
 - `<code>` in headings are incorrectly sized (such as on the v4 migration page)

---

## Issue 5 - Dark mode
[optional]
This is a nice to have to be tackled down the line. 

---

## Issue 6 - Individual page audit
This would be a big issue, maybe some subtasks within the issue for each page. 
This would be, systematically going through each page and updating explanations, checking code snippets are correct, etc


---

I think that's it from me!


Side question: You mentioned that valtio has a valtio website team. How is that currently structured?

**Reply by dai-shi** [reply]
> I'm not sure how you would like to handle the project management side of this

Good question. Basically, I want someone or a group of people to handle this.
I will be able to review changes from the correctness point of view.

> perhaps one issue (tagged as "documentation") based on some of the key areas below?

Yeah, creating issues and then PRs would be good. Would be nice if PRs are not too big, so that review can go gradually.

> Side question: You mentioned that valtio has a valtio website team. How is that currently structured?

I made a private group in Discord and invited who raised their hands. In the end, there are a few active members. We could do it, or just here in public is fine.

**Reply by chrisk-7777** [reply]
Great, that all makes sense. Thank you. 

I'll split up the above into individual issues, and will attempt to keep them small and tagged. 
My aim is to get it the issues organised over the next few days, then once that is done, I will then start to work through them. 

I will definitely need to check in with correctness, and probably thoughts around that top level structure to begin with.

**Reply by dbritto-dev** [reply]
Yeah, I agree with you folks.

**Reply by dai-shi** [reply]
> thoughts around that top level structure to begin with.

Looks familiar (with Jotai's one), so no surprising.
One thing to note is that we prefer to keep `readme.md` as is (or a little slimmer), so one question is if we expect people to already read it. maybe not, so probably there are some overlap between readme.md and introduction/getting-started. Will see.

**Reply by chrisk-7777** [reply]
I have created the following issues:

 - [x] https://github.com/pmndrs/zustand/issues/1215
 - [x] https://github.com/pmndrs/zustand/issues/1216
 - [x] https://github.com/pmndrs/zustand/issues/1217
 - [x] https://github.com/pmndrs/zustand/issues/1218
 - [x] https://github.com/pmndrs/zustand/issues/1219
 - [x] https://github.com/pmndrs/zustand/issues/1220

I will begin with a PR for 1215 to get the top structure setup. 
This may be a few days, but I'll aim to keep the momentum going

**Reply by lowellwashere2020** [reply]
Are you y'all still looking for people to help? I'd love to take part.

**Reply by chrisk-7777** [reply]
@lowellwashere2020 would you be willing to tackle #1216 ? Basically going through and updating any broken links? There are a few examples listed in the issue itself but I'm sure there are quite a few more. 

If you look at that, I'll start looking at #1217 tonight.

**Reply by dai-shi** [reply]
thanks guys! btw, i will be away for a couple of days. will check things out when i'm back.

**Reply by dai-shi** [reply]
Can anyone fix links in wiki? https://github.com/pmndrs/zustand/wiki/3rd-Party-Libraries and other pages.

**Reply by dbritto-dev** [reply]
@dai-shi sure I can take a look of that

**Reply by lowellwashere2020** [reply]
> @lowellwashere2020 would you be willing to tackle #1216 ? Basically going through and updating any broken links? There are a few examples listed in the issue itself but I'm sure there are quite a few more.
> 
> If you look at that, I'll start looking at #1217 tonight.


@chrisk-7777  Yes. I'll start on #1216 later today.

**Reply by Exalted100** [reply]
Hello. I'm interested in helping, if there are any issues I could pick up.

**Reply by chrisk-7777** [reply]
@Exalted100 awesome! 
I think #1218 + #1219 would be related to the parent style site (I need to confirm), but would you be able to do a few of the subtasks on #1220 ?

So we don't overlap, would you be able to drop a comment on #1220 saying "I'm working on subtask 'testing'" (as an example)?

**Reply by Exalted100** [reply]
Alright.

**Reply by dbritto-dev** [reply]
@dai-shi broken links in wiki were fixed and should works now.

**Reply by dbritto-dev** [reply]
@dai-shi let's add a section to explain how to contribute to zustand.

**Reply by dbritto-dev** [reply]
Hey guys, do someone know how to build the docs? I have tried to install npm packages but I got an error instead. 

![Screenshot from 2022-08-26 17-38-27](https://user-images.githubusercontent.com/1697714/186999398-586de835-0795-4c2d-89e7-d9213b4c666e.png)

**Reply by Exalted100** [reply]
I used `yarn install` and that worked fine.

**Reply by dai-shi** [reply]
> @dai-shi let's add a section to explain how to contribute to zustand. 

How about creating ./docs/README.md?
Or is it not only about docs?
In any case, we want to reduce top readme.md, not adding anything.

**Reply by dbritto-dev** [reply]
@dai-shi it's just about the docs for now. What about ./docs/guides/(contributing or contribute).md? And then we can add a section for docs for now. In the future we can add a section for other things.

**Reply by dai-shi** [reply]
It doesn't seem to fall into guides. Can it be ./docs/CONTRIBUTING.md?

**Reply by dbritto-dev** [reply]
@dai-shi yeah, that works as well

**Reply by dbritto-dev** [reply]
Hey folks, would you mind reviewing this PR (https://github.com/pmndrs/zustand/pull/1236)?

**Reply by dbritto-dev** [reply]
Hey folks, what you think about this (https://github.com/pmndrs/zustand/pull/1239)?. I want to add jotai and recoil as part of that section.

**Reply by Exalted100** [reply]
Hello, would anyone mind reviewing this PR: [#1244](https://github.com/pmndrs/zustand/pull/1244)

**Reply by dbritto-dev** [reply]
Hey folks, would someone mind reviewing this PR (https://github.com/pmndrs/zustand/pull/1239) ?

**Reply by chrisk-7777** [reply]
Looks good @dbritto-dev I just made some minor grammatical notes. 
You're crushing it 👍

**Reply by dbritto-dev** [reply]
@chrisk-7777 you rocks, thanks a lot. I'll apply your comments.

**Reply by dbritto-dev** [reply]
Hey folks, I'm back and I'll start working with gotchas wiki page and then I'll back to work on fixing some styling issues.

**Reply by ghost** [reply]
Hi all, very interested in helping out with the documentation for Zustand in my spare time when I'm not studying. 
Let's discuss further!

**Reply by Rec0iL99** [reply]
Hey folks, I would also like to help out with the documentation!

**Reply by chrisk-7777** [reply]
Awesome, thanks @Rec0iL99 . 
There are a few issues that are labeled "documentation" and not yet assigned, like a newly added one about slices. Is that something you would want to tackle?

**Reply by Rec0iL99** [reply]
Definitely @chrisk-7777, I will give it a try.

**Reply by Rec0iL99** [reply]
@chrisk-7777 @dai-shi

 I added a PR #1344 for the slices pattern issue #1253. Please do have a look.

Thanks.

**Reply by chrisk-7777** [reply]
Looks good, thanks @Rec0iL99 
Once merged, we can close off #1253 🙌

**Reply by dai-shi** [reply]
Can anyone look into #1365?

**Reply by Rec0iL99** [reply]
Sure @dai-shi, I will get it done

**Reply by chrisk-7777** [reply]
@guenyoo following up on the conversation from https://github.com/pmndrs/zustand/discussions/1378#discussioncomment-3948264

I'm more than happy to help get you started with the docs. 
Let me know what you need help with and I'll make sure to answer. To start with though, checkout this repo and look through the files located in docs. This will give you a bit of a start. 

Any specifics, please don't hesitate to @ me ! 

Thanks for offering to help :)

**Reply by guenyoo** [reply]
Hey @chrisk-7777, I'm quite new to contributing on github. Is there a guide on how to contribute somewhere? Play rules, what to look out for etc.? I would gladly like to correct the example I've mentioned in my original issue #1378 but I'm not quite sure how I'll be able to do that. Could you perhaps point me in the right direction?

**Reply by chrisk-7777** [reply]
Hey @guenyoo !
There is no contributing guide at the moment, but I think common sense when it comes to "play rules" is the way to go. 

In terms of actually jumping in and writing up some docs, I would do the following: 
 - Fork the Zustand repo into your own Github account
 - Locally, clone down your fork
 - Separately, clone https://github.com/pmndrs/website - this runs most of the doc websites under the pmndrs banner, including react three fiber and (eventually) zustand
 - Now, you should have two repo's locally
 - Inside the website folder, run `npm install` and then `npm run dev`, this will launch the docs site locally, that you should be able to open in the browser and see the various doc sites
 - One little catch here is that the docs site currently reads only from Github, not locally. So you can do the following (without committing it any where):
   - Inside website, open `src/data/libraries.ts` 
   - Within the Zustand key, change `docs: 'pmndrs/zustand/main/docs,'` to `docs: '[username]/zustand/[test-branch]/docs',`
   - For example, I have it set to `docs: 'chrisk-7777/zustand/docs-test/docs',`
   - Now, inside your Zustand fork, make the appropriate changes to the .md files in the `/docs` folder
   - Push those changes up to _your_ fork on a branch such as `docs-test`(this should match whatever you set it to above in `libraries.ts`
   - Restart the website locally (control + c -> `npm run dev`)
   - Visit the Zustand docs locally, and you should see the content you just pushed up
 - Once you are happy with your changes, commit them to a real branch and push up to your fork
   - I don't believe there are formal naming conventions for branches
   - Commit messages seem to follow a commitzen style commit message (eg `docs(issue-no): some description`)
 - Jump back to the official repo (this one) and raise a PR from the branch you just pushed up
 - ~Add myself or dai-shi as a reviewer or both. I'll do an initial review but dai-shi will have the final call~

Hopefully that helps! Let me know if you get stuck at any of the steps

**Reply by guenyoo** [reply]
Hey @chrisk-7777 thanks a lot for the detailed guide. I had already started and I have created a Pull Request, however, I can't seem to assign you as reviewers (there's no button to click for me). What am I doing wrong? This is the PR: https://github.com/pmndrs/zustand/pull/1388

![image](https://user-images.githubusercontent.com/15145968/197984192-df2ba042-43ae-47be-b37e-d85ff8560046.png)

**Reply by chrisk-7777** [reply]
Oh, it might be just a permissions thing - leave it without a reviewer in that case :) 
I'll update my notes above

**Reply by dai-shi** [reply]
Can someone take this? #1389

**Reply by Rec0iL99** [reply]
@dai-shi @chrisk-7777 Sorry for the delay in making a pr for #1365. 

pr #1404 should fix it.

**Reply by Rec0iL99** [reply]
> Can someone take this? #1389

opened pr for this #1412

**Reply by Rec0iL99** [reply]
https://docs.pmnd.rs/zustand/getting-started/concepts

@dai-shi @chrisk-7777 what would come in this doc? I couldn't find anything in the readme.

**Reply by chrisk-7777** [reply]
@Rec0iL99 The intention is to have it similar to Jotai. 

Introduction would be a quick overview dot point list of what Zustand is, and how to navigate the docs. 
Concepts would be the fundamental mechanics that make up Zustand. What problem does it solve? Why would you reach for this library? 

Jotai concepts page: https://jotai.org/docs/basics/concepts

**Reply by chrisk-7777** [reply]
In saying that, there is probably a really fine difference between Introduction and Concepts, and it might be less mental overhead just to merge the two, with a "concepts" section within Introduction. 

Edit: If you haven't already started looking at it @Rec0iL99 , I will tidy up the intro / concepts page in the next few days. It would be good to get that first welcome to the library squared away

**Reply by Rec0iL99** [reply]
> In saying that, there is probably a really fine difference between Introduction and Concepts, and it might be less mental overhead just to merge the two, with a "concepts" section within Introduction. 
> 
> 
> 
> Edit: If you haven't already started looking at it @Rec0iL99 , I will tidy up the intro / concepts page in the next few days. It would be good to get that first welcome to the library squared away 

No, I haven't started anything. You can go ahead with it.

**Reply by sewera** [reply]
Hi @chrisk-7777, I've got some additional steps that I had to take for the docs pages to work. You can edit your comment to add those if you find them useful.
I will list them under the quoted steps that preceded mine:

> * Separately, clone https://github.com/pmndrs/website - this runs most of the doc websites under the pmndrs banner, including react three fiber and (eventually) zustand

You also have to checkout the `docs` branch in the `pmndrs/website` repo.

> * Inside the website folder, run `npm install` and then `npm run dev`, this will launch the docs site locally, that you should be able to open in the browser and see the various doc sites

There is a `yarn.lock` file, so it's a better idea to run `yarn` and then `yarn dev`, as it's written in `readme.md`.

> * One little catch here is that the docs site currently reads only from Github, not locally. So you can do the following (without committing it any where):
>       
>   * Inside website, open `src/data/libraries.ts`
>   * Within the Zustand key, change `docs: 'pmndrs/zustand/main/docs,'` to `docs: '[username]/zustand/[test-branch]/docs',`
>   * For example, I have it set to `docs: 'chrisk-7777/zustand/docs-test/docs',`
>   * Now, inside your Zustand fork, make the appropriate changes to the .md files in the `/docs` folder
>   * Push those changes up to _your_ fork on a branch such as `docs-test` (this should match whatever you set it to above in `libraries.ts`

I also had to remove the `temp` directory in the `website` repo (`rm -rf ./temp`), because it seems that NextJS caches the Markdown docs from Github there.

Note that I launched the dev server **before** changing the docs source to my fork of Zustand, so your mileage may vary there.

> * [...]
>   * Restart the website locally (control + c -> `npm run dev`)
>   * Visit the Zustand docs locally, and you should see the content you just pushed up

If that doesn't work, try re-deleting the `temp` directory, as in the previous step. It will take some time to fetch the changes from Github and generate a page, so be patient here.

**Reply by dai-shi** [reply]
Hi @sewera and others,

When you have time, would you please look at docs marked as "under construction"?
```
docs/getting-started/concepts.md
8:  ⚠️ This doc is still under construction. https://github.com/pmndrs/zustand/discussions/1033

docs/getting-started/introduction.mdx
8:  ⚠️ This doc is still under construction.

docs/recipes/recipes.mdx
7:⚠️ This doc is still under construction. https://github.com/pmndrs/zustand/discussions/1033
```

There are no special requirements for those. Feel free to rewrite them completely, or even delete them.

**Reply by sewera** [reply]
I will most probably look into it on Thursday

**Reply by chrisk-7777** [reply]
Added PR here: https://github.com/pmndrs/zustand/pull/1549
Marked @sewera as reviewer.

**Reply by sewera** [reply]
To avoid unnecessary back-and-forth, I've created #1559.
I merged @chrisk-7777's ideas and mine into one PR.
Chris and @dai-shi, you can review those changes.
I tried to describe them as well as I could in the PR description.

**Reply by chrisk-7777** [reply]
I've closed #1549 in favor of #1559

**Reply by holgergp** [reply]
@chrisk-7777 The link to the docs (now?) seems to be in `footer.js`. Am I looking at the right one?

**Reply by sewera** [reply]
I'm not sure what do you mean, but the link to the docs is <https://docs.pmnd.rs/zustand/>.

The website src code is in [pmndrs/website](https://github.com/pmndrs/website) on branch `docs`.

The docs src code is in this repo in the `docs` directory.

@dai-shi, I think the website (<https://docs.pmnd.rs/zustand/>) is not updated to the latest version (from my previous experience, the Next.js cache has to be invalidated/deleted).

**Reply by holgergp** [reply]
Ah! The docs branch! Thank you! I was looking at main. Sorry!

**Reply by dai-shi** [reply]
> I'm not sure what do you mean, but the link to the docs is https://docs.pmnd.rs/zustand/.
> 
> The website src code is in [pmndrs/website](https://github.com/pmndrs/website) on branch `docs`.
> 
> The docs src code is in this repo in the `docs` directory.
> 
> @dai-shi, I think the website (https://docs.pmnd.rs/zustand/) is not updated to the latest version (from my previous experience, the Next.js cache has to be invalidated/deleted).

If the problem persists, you may want to open an issue there. I'm not very familiar with how it's working.

**Comment by dai-shi** [comment]
The typescript guide is getting bigger. Do we have ideas to split it into multiple files? Like for basic usage, advanced usage, for middleware devs, and detailed internals. 

Cc @dbritto-dev @sewera @chrisk-7777 @devanshj

**Reply by dbritto-dev** [reply]
Yeah I like the idea of having basic, advance usage, middlewares and pitfalls guides

**Reply by devanshj** [reply]
I think it's better if the guide remains a single succint cheatsheet that serves as a go-to point for dealing with TypeScript, too much fragmentation won't be good, besides we already have sections. But that's just my suggestion, I don't have any objections as such on spliting it up.

**Reply by dai-shi** [reply]
Maybe, separating guide and tutorial?

**Reply by devanshj** [reply]
Which section (or sections) are you referring to as "tutorial"?

**Reply by dai-shi** [reply]
I didn't mean anything specific. Tutorial would be something like with steps?

**Reply by devanshj** [reply]
Yes but we purposely don't have anything much tutorialy (as I also mention in #1581).

**Reply by dbritto-dev** [reply]
A tutorial would be like how to use the vanilla version then end up with react version, how to use without slices version then end up with slices version, how you can create your own middleware, which middlewares are available, how you can integrate it with other packages like jotai, etc

Each tutorial should start with typescript, then after each code block with add summary/toggle section to show the code without typescript

**Reply by sewera** [reply]
I think the current Typescript Guide docs page is okay.
Because for the majority of this project's life I was only the user of this library,
and I've come from Redux, like many React veterans, I guess,
I try to navigate the Zustand docs like the Redux ones.

In the [usage with TypeScript page of the Redux docs](https://redux.js.org/usage/usage-with-typescript) you can see that
they are using the _tutorial_ style extensively.

That's why, when I look for the slices pattern,
I ended up on the [Slices pattern Zustand doc page](https://docs.pmnd.rs/zustand/guides/slices-pattern), no problem.
When I looked how to **type** the slices pattern,
I ended up in the [Slices pattern section of the TypeScript guide](https://docs.pmnd.rs/zustand/guides/typescript#slices-pattern), also no problem.

One thing I'd maybe change is renaming "Slices pattern" section
to "Typing the slices pattern" to match the Redux style.

However, I'm aware that not everyone has a similar background
and way of thinking as me, but I wanted to chip in with my opinion :)

**Reply by dbritto-dev** [reply]
For redux users would be good to have a tutorial about migrating redux/rtk to zustand following/using the code on the redux tutorial

**Reply by sewera** [reply]
I was more in a "I used Redux in the past,
I want to use Zustand for my next project" kind of place.
That's why I didn't mention a tutorial
about migration from Redux to Zustand.

I think it may be a useful addition to the docs.
I found [this part of docs](https://docs.pmnd.rs/zustand/guides/flux-inspired-practice#flux-like-patterns-/-%22dispatching%22-actions) useful for that.

From my perspective, as a kind of user mentioned above,
my previous point of the docs being okay as they are stands.

**Reply by dai-shi** [reply]
Thanks everyone.
To clarify, my original point was actually simple. `typescript.md` is bigger than other files. If it's not too confusing, it's fine.

Anyways, it has been a good discussion. What should be our action items?

**Reply by sewera** [reply]
I'm all for the comment about splitting the TS guide to basic and advanced usage ([this comment](https://github.com/pmndrs/zustand/pull/1581#issuecomment-1407604175))

**Reply by holgergp** [reply]
> I was more in a "I used Redux in the past, I want to use Zustand for my next project" kind of place. That's why I didn't mention a tutorial about migration from Redux to Zustand.
> 
> I think it may be a useful addition to the docs. I found [this part of docs](https://docs.pmnd.rs/zustand/guides/flux-inspired-practice#flux-like-patterns-/-%22dispatching%22-actions) useful for that.
> 
> From my perspective, as a kind of user mentioned above, my previous point of the docs being okay as they are stands.

As I have used Redux quite extensively in the past, I like the idea of having something dedicated that guides users from known redux concepts and recipes, to the "zustand"-way. Might (or might not )be a good fit for the docs. Might also be nice from a seo point of view.
If I find the time in coming weeks, I could try and scribble something. Even if it is just something to sort my own understandings of the concepts.

**Reply by dai-shi** [reply]
> As I have used Redux quite extensively in the past, I like the idea of having something dedicated that guides users from known redux concepts and recipes, to the "zustand"-way.

It would be super nice! We are open for it.

**Comment by Vishrut19** [comment]
Hey! How can I contribute to your project?

**Reply by dai-shi** [reply]
@dbritto-dev @sewera @chrisk-7777 any thoughts?

**Reply by sewera** [reply]
@Vishrut19 If you meant what do you have to _technically_ do to contribute, look at the `CONTRIBUTING.md` file at the root of the Zustand repo.

**Reply by chrisk-7777** [reply]
Thanks for the offer to help out @Vishrut19 ! 🎉 

The good news is we reached a point where the core docs are at a place that we felt comfortable releasing them as an actual site. The original list of tasks can be found under this comment above (https://github.com/pmndrs/zustand/discussions/1033#discussioncomment-3463014)

Of course, there is always room for improvement - there is always spots that can be iterated on and made better. 
Unfortunately, at this stage the list of things to do are less clear, however if you're keen to get stuck in there are a few options:

**Option 1**: Being you have a fresh set of eyes, you may be able to spot introductory issues that we have missed. So you could read the docs from the start and ask the question "is this clear enough for someone new to Zustand to get started?" Are there any assumptions in knowledge, any steps skipped? 

**Option 2**:  Pick a page (or collection of pages) and look for general grammar issues, spelling mistakes, or generally unclear explanations. 

For option 1, I would encourage raising it here first as a talking point, so we can discuss before jumping into changes, to make sure it all aligns. 

For option 2, with any clear grammar/spelling issues, please feel free to open a PR directly. 


If you get stuck running the docs locally, sing out, and we'll give you a hand. As sewara mentioned, the CONTRIBUTING file is a great place to get setup. 

Thanks again 👍

**Comment by dai-shi** [comment]
Do we need recipes.mdx? It's originally a copy of the readme.md section and maintained separately.

An idea would be if we have unique descriptions in recipes, move them to guides.

On the other hand, I personally like "recipes" if we have such contents. Something like this in Jotai: https://jotai.org/docs/recipes/atom-creators

**Reply by dbritto-dev** [reply]
@dai-shi I'm down with that, could help us answering the same question many times. I've been trying to just pointing to my previous answers on discussions and creating some codesandbox demos for people

**Reply by dai-shi** [reply]
Hi @dbritto-dev, can you work on it please? It has more priority than https://github.com/pmndrs/zustand/discussions/1033#discussioncomment-7107315 below.

- Move some contents in `recipes.mdx` that are not duplicated with `readme.md` to somewhere else in `docs/`.
- Remove `recipes.md`

**Reply by dbritto-dev** [reply]
@dai-shi here you go -> https://github.com/pmndrs/zustand/pull/2085

**Comment by dai-shi** [comment]
An idea from @pmeaney
https://github.com/pmndrs/zustand/discussions/2074#discussioncomment-7103853

**Reply by dbritto-dev** [reply]
I'm into that

**Reply by pmeaney** [reply]
@dai-shi @dbritto-dev  🙇 Happy to help if I can.  Currently focused on picking up Nextjs (my current learning binge) + Zustand (my upcoming learning binge).  I haven't dug into Zustand yet, but the one thing I wanted to suggest was just that markdown data table layout for showing demo examples (similar to KinglyJS's github readme examples)

**Reply by pmeaney** [reply]
@dai-shi @dbritto-dev 
Example idea:
Ask ChatGTP something like this:
A. Please generate a Zustand finite state machine which powers a NextJS UI, for a oauth2 sign-up and sign-in system. (common & important use case for making an app)  (Ask for it to elaborate on the main components.  Bring them into a project...)

< Nah-- it's out of scope unless it's simple and local B. From here, potentially provided would be Dockerized (docker-compose) templates for underlying systems (redis--  sessions, postgres-- user system), etc. OR local .jsonfile /  Window: localStorage property  / sqlite.           />

Place A & potentially A+B into an example.  A--> Codepen.  A+B -->github...

__ input above___

C. output to zustand docs:   .md demo datatable row: A. diagram, B. live .gif demo, C. code examples: codepen + github repo template

__________________________
Next:

Ask ChatGPT-- How do I integrate my AuthFSM with my ... PrototypeUXPageFSM? ...Make some basic UX functionality: a kanban board? Something like that.  Drag & Drop... or an interactive Data visualization...

AuthFSM -> somePrototypeFSM... how do I make them work together yet stay decoupled?

**Reply by dbritto-dev** [reply]
@pmeaney @armaandh  here you go -> https://github.com/pmndrs/zustand/discussions/2489

**Comment by mcqj** [comment]
For the JS developers who don't embrace Typescript, the documentation is difficult to follow.
The norm on most sites seems to be to have the ability to switch examples between JS and TS and it would be a great addition to the documentation.

**Reply by dbritto-dev** [reply]
@mcqj yeah, we are working on that
